### PR TITLE
Removing CosmosContainers and CosmosDatabases plural type

### PR DESCRIPTION
**Pull request by kirankumarkolli**
- [x] Flatten plural types (CosmosDatabases & CosmosContaiers).
- [x] Remove statefull dictionary for databases and containers
- [x] [] indexer replaced by GetDatabase and GetContainer
- [x] Shortcut in client with GetContainer(dbName, containerName)

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Closing issues
related to #333

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainers.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Azure.Cosmos
     /// For instance, do not call `containers.GetContainersIterator()` before every single `item.read()` call, to ensure the container exists;
     /// do this once on application start up.
     /// </summary>
-    public abstract class CosmosContainers
+    public abstract partial class CosmosDatabase
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainers.cs]: We should at least merge the contract classes into the correct file so it's clear what the final contract for each object.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainers.cs]: #313

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 2]
```diff
@@ -246,6 +232,28 @@ public virtual Task<CosmosAccountSettings> GetAccountSettingsAsync()
             return ((IDocumentClientInternal)this.DocumentClient).GetDatabaseAccountInternalAsync(this.ClientOptions.AccountEndPoint);
         }
 
+        /// <summary>
+        /// Get cosmos container proxy. 
+        /// Proxy existence doesn't guarantee either database or container existence.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: This should be a `<remark>` ?

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 3]
```diff
@@ -246,6 +232,28 @@ public virtual Task<CosmosAccountSettings> GetAccountSettingsAsync()
...
+        /// </summary>
+        /// <param name="databaseId">cosmos database name</param>
+        /// <param name="containerId">comsos container name</param>
+        /// <returns>Cosmos container proxy</returns>
+        public virtual CosmosContainer GetContainer(string databaseId, string containerId)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Shouldn't this be on the database only? Are users expected to use this one or `GetDatabase(...).GetContainer(...)`

File: Microsoft.Azure.Cosmos/src/Resource/Database/CosmosDatabasesCore.cs [hunk 4]
```diff
@@ -5,33 +5,39 @@
...
-    using Microsoft.Azure.Cosmos.Internal;
     using Microsoft.Azure.Documents;
 
     /// <summary>
     /// Operations for creating new databases, and reading/querying all databases
     ///
     /// <see cref="CosmosDatabase"/>for reading, replacing, or deleting an existing container;
     /// </summary>
-    internal class CosmosDatabasesCore : CosmosDatabases
+    public partial class CosmosClient
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Database/CosmosDatabasesCore.cs]: File name does not match class name?

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs (hunk 3 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Its a short-cut for good compact code.

File: Microsoft.Azure.Cosmos/src/Resource/Database/CosmosDatabasesCore.cs (hunk 4 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Database/CosmosDatabasesCore.cs]: Tracked part of  #313

**Pull request merged**